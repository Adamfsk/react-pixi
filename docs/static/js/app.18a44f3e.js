(window.webpackJsonp=window.webpackJsonp||[]).push([[1],{"./.docz/app/db.json":function(e){e.exports={config:{title:"React Pixi",description:"Write PIXI applications using React declarative style",themeConfig:{colors:{primary:"tomato"}},ordering:"ascending",version:"0.1.2",repository:"https://github.com/inlet/react-pixi",propsParser:!1,hashRouter:!0,dest:"docs"},entries:{"README.mdx":{name:"About React Pixi",order:0,route:"/",id:"075c8e9dd56c748f3da128a33861a770",filepath:"README.mdx",link:"https://github.com/inlet/react-pixi/edit/master/README.mdx",slug:"readme",menu:null,headings:[{depth:1,slug:"react-fiber-renderer-for-pixi",value:"React fiber renderer for pixi"},{depth:2,slug:"install",value:"Install"},{depth:3,slug:"node",value:"Node"},{depth:3,slug:"browser",value:"Browser"},{depth:2,slug:"usage",value:"Usage"},{depth:4,slug:"with-react-dom",value:"With react dom"},{depth:4,slug:"without-react-dom",value:"Without react dom"},{depth:2,slug:"examples",value:"Examples"},{depth:2,slug:"custom-components",value:"Custom components"},{depth:2,slug:"access-pixi-application",value:"Access pixi application"},{depth:3,slug:"via-render-props-function-as-child",value:"Via render props function as child"},{depth:3,slug:"via-higher-order-component-withpixiapp",value:"Via higher order component withpixiapp"},{depth:2,slug:"stage",value:"Stage"},{depth:3,slug:"props",value:"Props"},{depth:2,slug:"components",value:"Components"}]},"src/components/BitmapText.mdx":{name:"BitmapText",order:10,menu:"Components",route:"/components/bitmap-text",id:"be3209d167db8b7dbffcef178bad87e7",filepath:"src/components/BitmapText.mdx",link:"https://github.com/inlet/react-pixi/edit/master/src/components/BitmapText.mdx",slug:"src-components-bitmap-text",headings:[{depth:1,slug:"bitmaptext",value:"Bitmaptext"},{depth:2,slug:"props",value:"Props"},{depth:2,slug:"usage",value:"Usage"},{depth:2,slug:"example",value:"Example"}]},"src/components/Container.mdx":{name:"Container",menu:"Components",route:"/components/container",order:1,id:"2e5bc2a2cd68ab9f0f8d3df1f4d34462",filepath:"src/components/Container.mdx",link:"https://github.com/inlet/react-pixi/edit/master/src/components/Container.mdx",slug:"src-components-container",headings:[{depth:1,slug:"container",value:"Container"},{depth:2,slug:"props",value:"Props"},{depth:2,slug:"usage",value:"Usage"},{depth:2,slug:"example",value:"Example"}]},"src/components/Graphics.mdx":{name:"Graphics",order:5,menu:"Components",route:"/components/graphics",id:"46f51752305a331c95688ef0dfbac9f1",filepath:"src/components/Graphics.mdx",link:"https://github.com/inlet/react-pixi/edit/master/src/components/Graphics.mdx",slug:"src-components-graphics",headings:[{depth:1,slug:"graphics",value:"Graphics"},{depth:2,slug:"props",value:"Props"},{depth:3,slug:"draw",value:"Draw"},{depth:2,slug:"usage",value:"Usage"},{depth:2,slug:"example",value:"Example"}]},"src/components/Mesh.mdx":{name:"Mesh",order:6,menu:"Components",route:"/components/mesh",id:"241af672f33b0e74e7740e76058da4b7",filepath:"src/components/Mesh.mdx",link:"https://github.com/inlet/react-pixi/edit/master/src/components/Mesh.mdx",slug:"src-components-mesh",headings:[{depth:1,slug:"mesh",value:"Mesh"},{depth:2,slug:"props",value:"Props"},{depth:2,slug:"usage",value:"Usage"},{depth:2,slug:"example",value:"Example"}]},"src/components/NineSlicePlane.mdx":{name:"NineSlicePlane",order:8,menu:"Components",route:"/components/mesh/nine-slice-plane",id:"6af4bc991617b1261af6cd7a06121fcf",filepath:"src/components/NineSlicePlane.mdx",link:"https://github.com/inlet/react-pixi/edit/master/src/components/NineSlicePlane.mdx",slug:"src-components-nine-slice-plane",headings:[{depth:1,slug:"ninesliceplane",value:"Ninesliceplane"},{depth:2,slug:"props",value:"Props"},{depth:2,slug:"usage",value:"Usage"},{depth:2,slug:"example",value:"Example"}]},"src/components/ParticleContainer.mdx":{name:"ParticleContainer",order:2,menu:"Components",route:"/components/particle-container",id:"b241ab7afa23042ecb1ca32bf94c7476",filepath:"src/components/ParticleContainer.mdx",link:"https://github.com/inlet/react-pixi/edit/master/src/components/ParticleContainer.mdx",slug:"src-components-particle-container",headings:[{depth:1,slug:"particlecontainer",value:"Particlecontainer"},{depth:2,slug:"props",value:"Props"},{depth:3,slug:"maxsize",value:"Maxsize"},{depth:3,slug:"properties",value:"Properties"},{depth:3,slug:"batchsize",value:"Batchsize"},{depth:3,slug:"autoresize",value:"Autoresize"},{depth:2,slug:"usage",value:"Usage"},{depth:2,slug:"example",value:"Example"}]},"src/components/Rope.mdx":{name:"Rope",order:7,menu:"Components",route:"/components/mesh/rope",id:"a42b1d058c5c14224da80287ee2a57cc",filepath:"src/components/Rope.mdx",link:"https://github.com/inlet/react-pixi/edit/master/src/components/Rope.mdx",slug:"src-components-rope",headings:[{depth:1,slug:"rope",value:"Rope"},{depth:2,slug:"props",value:"Props"},{depth:2,slug:"usage",value:"Usage"},{depth:2,slug:"example",value:"Example"}]},"src/components/Sprite.mdx":{name:"Sprite",order:3,menu:"Components",route:"/components/sprite",id:"a85d3bf3e3eb2bc95c10eef22bbe23cf",filepath:"src/components/Sprite.mdx",link:"https://github.com/inlet/react-pixi/edit/master/src/components/Sprite.mdx",slug:"src-components-sprite",headings:[{depth:1,slug:"sprite",value:"Sprite"},{depth:2,slug:"props",value:"Props"},{depth:2,slug:"usage",value:"Usage"},{depth:2,slug:"example",value:"Example"}]},"src/components/Text.mdx":{name:"Text",order:9,menu:"Components",route:"/components/text",id:"45ae8ab60e42d20e40bf547ae288fbc2",filepath:"src/components/Text.mdx",link:"https://github.com/inlet/react-pixi/edit/master/src/components/Text.mdx",slug:"src-components-text",headings:[{depth:1,slug:"text",value:"Text"},{depth:2,slug:"props",value:"Props"},{depth:2,slug:"style-props",value:"Style props"},{depth:2,slug:"usage",value:"Usage"},{depth:2,slug:"example",value:"Example"}]},"src/components/TilingSprite.mdx":{name:"TilingSprite",order:4,menu:"Components",route:"/components/tiling-sprite",id:"96901b8d151744c970684916804051db",filepath:"src/components/TilingSprite.mdx",link:"https://github.com/inlet/react-pixi/edit/master/src/components/TilingSprite.mdx",slug:"src-components-tiling-sprite",headings:[{depth:1,slug:"tilingsprite",value:"Tilingsprite"},{depth:2,slug:"props",value:"Props"},{depth:2,slug:"usage",value:"Usage"},{depth:2,slug:"example",value:"Example"}]},"src/stage/Stage.mdx":{name:"Stage",order:1,route:"/stage",id:"d912e9fd43740cdfc8c6e92ca9e55715",filepath:"src/stage/Stage.mdx",link:"https://github.com/inlet/react-pixi/edit/master/src/stage/Stage.mdx",slug:"src-stage-stage",menu:null,headings:[{depth:1,slug:"stage",value:"Stage"},{depth:2,slug:"props",value:"Props"},{depth:2,slug:"usage",value:"Usage"},{depth:2,slug:"example",value:"Example"}]}}}},"./.docz/app/index.jsx":function(e,t,n){"use strict";n.r(t);var s=n("./node_modules/react/index.js"),o=n.n(s),a=n("./node_modules/react-dom/index.js"),p=n.n(a),i={"README.mdx":function(){return n.e(2).then(n.bind(null,"./README.mdx"))},"src/components/BitmapText.mdx":function(){return Promise.all([n.e(0),n.e(3)]).then(n.bind(null,"./src/components/BitmapText.mdx"))},"src/components/Container.mdx":function(){return Promise.all([n.e(0),n.e(4)]).then(n.bind(null,"./src/components/Container.mdx"))},"src/components/Graphics.mdx":function(){return Promise.all([n.e(0),n.e(5)]).then(n.bind(null,"./src/components/Graphics.mdx"))},"src/components/Mesh.mdx":function(){return Promise.all([n.e(0),n.e(6)]).then(n.bind(null,"./src/components/Mesh.mdx"))},"src/components/NineSlicePlane.mdx":function(){return Promise.all([n.e(0),n.e(7)]).then(n.bind(null,"./src/components/NineSlicePlane.mdx"))},"src/components/ParticleContainer.mdx":function(){return Promise.all([n.e(0),n.e(8)]).then(n.bind(null,"./src/components/ParticleContainer.mdx"))},"src/components/Rope.mdx":function(){return Promise.all([n.e(0),n.e(9)]).then(n.bind(null,"./src/components/Rope.mdx"))},"src/components/Sprite.mdx":function(){return Promise.all([n.e(0),n.e(10)]).then(n.bind(null,"./src/components/Sprite.mdx"))},"src/components/Text.mdx":function(){return Promise.all([n.e(0),n.e(11)]).then(n.bind(null,"./src/components/Text.mdx"))},"src/components/TilingSprite.mdx":function(){return Promise.all([n.e(0),n.e(12)]).then(n.bind(null,"./src/components/TilingSprite.mdx"))},"src/stage/Stage.mdx":function(){return Promise.all([n.e(0),n.e(13)]).then(n.bind(null,"./src/stage/Stage.mdx"))}},r=n("./.docz/app/root.jsx"),l=[],c=[],d=function(){return c.forEach(function(e){return e&&e()})},m=document.querySelector("#root");(function(){var e=arguments.length>0&&void 0!==arguments[0]?arguments[0]:r.a;l.forEach(function(e){return e&&e()}),p.a.render(o.a.createElement(e,{imports:i}),m,d)})(r.a)},"./.docz/app/root.jsx":function(e,t,n){"use strict";(function(e){var s=n("./node_modules/react/index.js"),o=n.n(s),a=n("./node_modules/react-hot-loader/index.js"),p=n("./node_modules/docz-theme-default/dist/index.m.js"),i=n("./.docz/app/db.json");t.a=Object(a.hot)(e)(function(e){var t=e.imports;return o.a.createElement(p.a,{db:i,imports:t,hashRouter:!0})})}).call(this,n("./node_modules/webpack/buildin/harmony-module.js")(e))},0:function(e,t,n){n("./node_modules/docz-core/node_modules/babel-polyfill/lib/index.js"),e.exports=n("./.docz/app/index.jsx")}},[[0,14,0]],[0,3,4,5,6,7,8,9,10,11,12,13,2]]);