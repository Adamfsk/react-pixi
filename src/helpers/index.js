export const isFunction = (...args) => {
  return args.reduce((a, b) => a && typeof b === 'function', true)
}

export const isObject = obj => Object.prototype.toString.call(obj) === '[object Object]'

export const hasKey = collection => {
  let coll = collection

  if (!Array.isArray(coll)) {
    if (isObject(collection)) {
      coll = Object.keys(collection)
    } else {
      throw new Error('collection needs to be an Array or Object')
    }
  }

  const index = {}
  coll.forEach(key => {
    index[key] = true
  })

  return key => typeof index[key] !== 'undefined'
}

// See https://github.com/michalochman/react-pixi-fiber/blob/a4dbddbef0ffbf0f563c64d30766ea28222a51ea/src/utils.js#L7
export const not = boolFn => (...args) => !boolFn(...args)

export const runningInBrowser = () => Boolean(window)
