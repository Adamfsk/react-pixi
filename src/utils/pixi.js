import * as PIXI from 'pixi.js'
import idx from 'idx'
import isNil from 'lodash/isNil'

/**
 * Parse PIXI point value to array of coordinates
 *
 * @param {*} value
 * @returns {Array}
 */
export function parsePoint(value) {
  let arr = []

  if (typeof value === 'undefined') {
    return arr
  } else if (typeof value === 'string') {
    arr = value.split(',')
  } else if (typeof value === 'number') {
    arr = [value]
  } else if (Array.isArray(value)) {
    arr = [...value]
  } else if (value !== null && typeof value === 'object') {
    const x = idx(value, _ => _.x) || 0
    const y = idx(value, _ => _.y) || 0
    arr = [x, y]
  } else {
    return arr
  }

  return arr.filter(p => !isNil(p) && !isNaN(p)).map(Number)
}

/**
 * Determine value is type of Point or ObservablePoint
 *
 * @param {*} value
 * @returns {boolean}
 */
export function isPointType(value) {
  return value instanceof PIXI.Point || value instanceof PIXI.ObservablePoint
}

/**
 * Event handlers
 *
 * @type {string[]}
 */
export const eventHandlers = [
  'click',
  'mousedown',
  'mousemove',
  'mouseout',
  'mouseover',
  'mouseup',
  'mouseupoutside',
  'tap',
  'touchstart',
  'touchmove',
  'touchend',
  'touchendoutside',
  'pointercancel',
  'pointerout',
  'pointerover',
  'pointertap',
  'pointerdown',
  'pointerup',
  'pointerupoutside',
  'pointermove',
  'rightclick',
  'rightdown',
  'rightup',
  'rightupoutside',
  'touchcancel',
  'touchendoutside',
  'touchmove',
  'touchstart',
]
