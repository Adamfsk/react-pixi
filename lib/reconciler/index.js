import "core-js/modules/es6.function.name";
import Reconciler from 'react-reconciler';
import emptyObject from 'fbjs/lib/emptyObject';
import { createElement } from '../utils/createElement';
import now from 'performance-now';
import pkg from '../../package.json';
var PixiFiber = Reconciler({
  appendInitialChild: function appendInitialChild(parent, child) {
    if (parent.appendChild) {
      parent.appendChild(child);
    } else {
      parent.document = child;
    }
  },
  createInstance: function createInstance(type, props, internalInstanceHandler) {
    return createElement(type, props, internalInstanceHandler);
  },
  createTextInstance: function createTextInstance(text, rootContainerInstance, internalInstanceHandler) {
    return text;
  },
  finalizeInitialChildren: function finalizeInitialChildren(wordElement, type, props) {
    return false;
  },
  getPublicInstance: function getPublicInstance(ins) {
    return ins;
  },
  prepareForCommit: function prepareForCommit() {// noop
  },
  prepareUpdate: function prepareUpdate(wordElement, type, oldProps, newProps) {
    return true;
  },
  resetAfterCommit: function resetAfterCommit() {// noop
  },
  resetTextContent: function resetTextContent(wordElement) {// noop
  },
  getRootHostContext: function getRootHostContext(rootInstance) {// You can use this 'rootInstance' to pass data from the roots.
  },
  getChildHostContext: function getChildHostContext() {
    return emptyObject;
  },
  shouldSetTextContent: function shouldSetTextContent(type, props) {
    return false;
  },
  now: now,
  useSyncScheduling: true,
  mutation: {
    appendChild: function appendChild(parent, child) {
      if (parent.appendChild) {
        parent.appendChild(child);
      } else {
        parent.document = child;
      }
    },
    appendChildToContainer: function appendChildToContainer(parent, child) {
      if (parent.appendChild) {
        parent.appendChild(child);
      } else {
        parent.document = child;
      }
    },
    removeChild: function removeChild(parent, child) {
      parent.removeChild(child);
    },
    removeChildFromContainer: function removeChildFromContainer(parent, child) {
      parent.removeChild(child);
    },
    inserBefore: function inserBefore(parent, child, beforeChild) {// noop
    },
    commitUpdate: function commitUpdate(instance, updatePayload, type, oldProps, newProps) {// noop
    },
    commitMount: function commitMount(instance, updatePayload, type, oldProps, newProps) {// noop
    },
    commitTextUpdate: function commitTextUpdate(textInstance, oldText, newText) {
      textInstance.children = newText;
    }
  }
});
export default PixiFiber;
export var VERSION = pkg.version;
export var PACKAGE_NAME = pkg.name;