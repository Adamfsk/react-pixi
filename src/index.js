import { TYPES } from './utils/element'
import { render } from './render'
import Stage from './stage'
import Provider from './stage/provider'

/**
 * -------------------------------------------
 * Public API
 * -------------------------------------------
 */

export { render, Stage, Provider }

export const BitmapText = TYPES.BitmapText
export const Container = TYPES.Container
export const Graphics = TYPES.Graphics
export const NineSlicePlane = TYPES.NineSlicePlane
export const ParticleContainer = TYPES.ParticleContainer
export const Sprite = TYPES.Sprite
export const Text = TYPES.Text
export const TilingSprite = TYPES.TilingSprite
export const Shape = TYPES.Shape
export const Mesh = TYPES.Mesh
export const Rope = TYPES.Rope
