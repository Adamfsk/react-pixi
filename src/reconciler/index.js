import Reconciler from 'react-reconciler'
import emptyObject from 'fbjs/lib/emptyObject'
import { createElement } from '../utils/createElement'
import now from 'performance-now'

import pkg from '../../package.json'

/**
 * The React Reconciler handling PIXI elements
 *
 * @type {Object}
 */
const PixiFiber = Reconciler({
  appendInitialChild(parent, child) {
    if (parent.appendChild) {
      parent.appendChild(child)
    } else {
      parent.document = child
    }
  },

  createInstance(type, props, internalInstanceHandler) {
    return createElement(type, props, internalInstanceHandler)
  },

  createTextInstance(text, rootContainerInstance, internalInstanceHandler) {
    return text
  },

  finalizeInitialChildren(wordElement, type, props) {
    return false
  },

  getPublicInstance(ins) {
    return ins
  },

  prepareForCommit() {
    // noop
  },

  prepareUpdate(wordElement, type, oldProps, newProps) {
    return true
  },

  resetAfterCommit() {
    // noop
  },

  resetTextContent(wordElement) {
    // noop
  },

  getRootHostContext(rootInstance) {
    // You can use this 'rootInstance' to pass data from the roots.
  },

  getChildHostContext() {
    return emptyObject
  },

  shouldSetTextContent(type, props) {
    return false
  },

  now,

  useSyncScheduling: true,

  mutation: {
    appendChild(parent, child) {
      if (parent.appendChild) {
        parent.appendChild(child)
      } else {
        parent.document = child
      }
    },

    appendChildToContainer(parent, child) {
      if (parent.appendChild) {
        parent.appendChild(child)
      } else {
        parent.document = child
      }
    },

    removeChild(parent, child) {
      parent.removeChild(child)
    },

    removeChildFromContainer(parent, child) {
      parent.removeChild(child)
    },

    inserBefore(parent, child, beforeChild) {
      // noop
    },

    commitUpdate(instance, updatePayload, type, oldProps, newProps) {
      // noop
    },

    commitMount(instance, updatePayload, type, oldProps, newProps) {
      // noop
    },

    commitTextUpdate(textInstance, oldText, newText) {
      textInstance.children = newText
    },
  },
})

export default PixiFiber

export const VERSION = pkg.version

export const PACKAGE_NAME = pkg.name
