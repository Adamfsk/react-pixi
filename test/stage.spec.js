import React from 'react'
import * as PIXI from 'pixi.js'
import renderer from 'react-test-renderer'
import { PixiFiber } from '../src/reconciler'
import { runningInBrowser } from '../src/helpers'
import { Stage, Container, Text } from '../src'
import { mockToSpy } from './__utils__/mock'

jest.mock('../src/helpers', () => ({
  ...require.requireActual('../src/helpers'),
  runningInBrowser: jest.fn(),
}))

jest.mock('../src/reconciler')

describe.only('stage', () => {
  beforeEach(() => {
    jest.resetAllMocks()
    mockToSpy('../src/reconciler')
    runningInBrowser.mockImplementation(() => true)
  })

  test('prevent mount when window is not found', () => {
    runningInBrowser.mockImplementation(() => false)
    expect(() => renderer.create(<Stage />)).toThrow('Cannot mount Stage, window object is not defined')
  })

  test('renders a canvas element', () => {
    const tree = renderer.create(<Stage />).toJSON()
    expect(tree).toHaveProperty('type', 'canvas')
    expect(tree).toMatchSnapshot()
  })

  test('renders null if view is passed in options', () => {
    const options = {
      view: document.createElement('canvas'),
    }
    const tree = renderer.create(<Stage options={options} />).toJSON()
    expect(tree).toBeNull()
  })

  test('invariant options.view', () => {
    const options = { view: 123 }
    expect(() => renderer.create(<Stage options={options} />).toJSON()).toThrow(
      'options.view needs to be a `HTMLCanvasElement`'
    )
  })

  test('passes props to canvas element', () => {
    const id = 'stage'
    const className = 'canvas__element'
    const style = { border: '1px solid red' }
    const dataAttr = 'something'
    const tree = renderer.create(<Stage className={className} id={id} style={style} data-attr={dataAttr} />).toJSON()
    expect(tree.props).toEqual({ className, id, style, 'data-attr': dataAttr })
  })

  test('does not pass reserved props to renderer canvas element', () => {
    const options = { backgroundColor: 0xff0000 }
    const tree = renderer.create(<Stage height={500} width={500} options={options} />).toJSON()
    expect(tree).toHaveProperty('type', 'canvas')
    expect(tree.props).toEqual({})
  })

  test('creates a PIXI.Application with passed options', () => {
    const el = renderer.create(<Stage width={100} height={50} options={{ backgroundColor: 0xff0000 }} />)
    const app = el.getInstance().app

    expect(app.stage).toBeInstanceOf(PIXI.Container)
    expect(app).toBeInstanceOf(PIXI.Application)
    expect(app._options).toMatchObject({
      backgroundColor: 0xff0000,
      width: 100,
      height: 50,
    })
  })

  test('resize renderer when dimensions change', () => {
    const el = renderer.create(<Stage width={100} height={100} />)
    const app = el.getInstance().app

    expect(app.renderer).toHaveProperty('width', 100)
    expect(app.renderer).toHaveProperty('height', 100)

    el.update(<Stage width={1000} height={100} />)
    expect(app.renderer).toHaveProperty('width', 1000)
    expect(app.renderer).toHaveProperty('height', 100)

    el.update(<Stage width={1000} height={1000} />)
    expect(app.renderer).toHaveProperty('width', 1000)
    expect(app.renderer).toHaveProperty('height', 1000)

    el.update(<Stage width={100} height={100} />)
    expect(app.renderer).toHaveProperty('width', 100)
    expect(app.renderer).toHaveProperty('height', 100)
  })

  test('call onMount()', () => {
    const spy = jest.fn()
    const el = renderer.create(<Stage onMount={spy} />)

    expect(spy).toHaveBeenCalledTimes(1)
    expect(spy.mock.calls[0]).toHaveLength(1)
    expect(spy.mock.calls[0][0]).toBeInstanceOf(PIXI.CanvasRenderer)
  })

  test('can be unmounted', () => {
    const el = renderer.create(<Stage />)
    const instance = el.getInstance()

    jest.spyOn(instance, 'componentWillUnmount')

    el.unmount()
    expect(instance.componentWillUnmount).toBeCalled()
  })

  test('call PixiFiber.createContainer on componentDidMount', () => {
    const el = renderer.create(<Stage />)
    const stage = el.getInstance().app.stage

    expect(PixiFiber.createContainer).toHaveBeenCalledTimes(1)
    expect(PixiFiber.createContainer).toHaveBeenCalledWith(stage)
  })

  test('call PixiFiber.updateContainer on componentDidMount', () => {
    PixiFiber.updateContainer.mockClear()

    const el = renderer.create(
      <Stage>
        <Text text="Hello World!" />
      </Stage>
    )

    const instance = el.getInstance()

    expect(PixiFiber.updateContainer).toHaveBeenCalledTimes(1)
    expect(PixiFiber.updateContainer).toHaveBeenCalledWith(<Text text="Hello World!" />, instance.mountNode, instance)
  })
})
