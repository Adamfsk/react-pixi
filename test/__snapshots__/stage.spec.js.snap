// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`stage prop types 1`] = `
Object {
  "children": [Function],
  "height": [Function],
  "onMount": [Function],
  "options": [Function],
  "raf": [Function],
  "renderOnComponentChange": [Function],
  "width": [Function],
}
`;

exports[`stage prop types 2`] = `
Object {
  "height": 600,
  "onMount": [Function],
  "raf": true,
  "renderOnComponentChange": true,
  "width": 800,
}
`;

exports[`stage renders a canvas element 1`] = `<canvas />`;
